framework/src/mfem/equation_systems/EquationSystem.C

```diff
@@ -578,44 +608,131 @@ TimeDependentEquationSystem::BuildBilinearForms()
 }
 
 void
-TimeDependentEquationSystem::FormLegacySystem(mfem::OperatorHandle & op,
-                                              mfem::BlockVector & truedXdt,
-                                              mfem::BlockVector & trueRHS)
+TimeDependentEquationSystem::BuildMixedBilinearForms()
 {
+  EquationSystem::BuildMixedBilinearForms();
+  // Register mixed bilinear forms. Note that not all combinations may
+  // have a kernel
 
-  // Allocate block operator
-  DeleteAllBlocks();
-  _h_blocks.SetSize(_test_var_names.size(), _test_var_names.size());
-  // Form diagonal blocks.
+  // Create mblf for each test/trial variable pair with an added kernel
+  for (const auto i : index_range(_test_var_names))
+  {
+    auto test_var_name = _test_var_names.at(i);
+    auto test_td_mblfs =
+        std::make_shared<Moose::MFEM::NamedFieldsMap<mfem::ParMixedBilinearForm>>();
+    for (const auto j : index_range(_trial_var_names))
+    {
+      const auto & trial_var_name = _trial_var_names.at(j);
+      auto td_mblf = std::make_shared<mfem::ParMixedBilinearForm>(_test_pfespaces.at(j),
+                                                                  _test_pfespaces.at(i));
+      td_mblf->SetAssemblyLevel(_assembly_level);
+      // Register MixedBilinearForm if kernels exist for it, and assemble
+      // kernels
+      if (_td_kernels_map.Has(test_var_name) &&
+          _td_kernels_map.Get(test_var_name)->Has(trial_var_name) &&
+          _time_derivative_map.getTimeDerivativeName(test_var_name) != trial_var_name)
+      {
+        // Apply all mixed kernels with this test/trial pair
+        ApplyDomainBLFIntegrators<mfem::ParMixedBilinearForm>(
+            trial_var_name, test_var_name, td_mblf, _td_kernels_map);
+      }
+      // Recover and scale integrators from the mblf acting on the time integral of the trial
+      // variable corresponding to coupled_var_name. This is to apply the dt*du/dt contributions
+      // from the operator on the trial variable in the implicit integration scheme
+      const auto & trial_var_time_integral_name =
+          _time_derivative_map.getTimeIntegralName(trial_var_name);
+      if (_mblfs.Has(test_var_name) && _mblfs.Get(test_var_name)->Has(trial_var_time_integral_name))
+      {
+        // Recover and scale integrators from mblf. This is to apply the dt*du/dt contributions
+        // from the operator on the trial variable in the implicit integration scheme
+        auto mblf = _mblfs.Get(test_var_name)->Get(trial_var_time_integral_name);
+        auto integs = mblf->GetDBFI();
+        auto b_integs = mblf->GetBBFI();
```

Comment:
longer variable names would help this be self documenting

Reply:
Updated to use more descriptive names, and split into a new `ScaleAndAddBLFIntegrators` templated method to re-use the same code for mixed and non-mixed bilinear forms